### DISCUSSION TITLE
Update only one field from record

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2610] [Q&A]
Does it update whole record or only changed fields?

```
record, err := app.Dao().FindRecordById("articles", "RECORD_ID")
if err != nil {
    return err
}

record.Set("title", "Lorem ipsum")
record.Set("active", true)
record.Set("someOtherField", 123)

if err := app.Dao().SaveRecord(record); err != nil {
    return err
}
```

**Comment by ganigeorgiev** [comment]
It will perform an `INSERT/UPDATE` query with all the record fields. But why does it matter in your case?

**Reply by Oggy4th** [reply]
I have to update statistics field on lot of records and I didn't want it to lock db and run in transaction, so I was searching if there was any possibility to update single field without overriding other fields and run it without transaction. I guess I can separate it to different collection and use relation.

**Reply by ganigeorgiev** [reply]
We usually submit all fields in order to apply normalization on updated fields.
One workaround for your case could be to create a custom model with only a subset of the fields and save it instead:

```go
type Article struct {
	models.BaseModel

	Title          string `db:"title" json:"title"`
	Active         bool   `db:"active" json:"active"`
	SomeOtherField int `db:"someOtherField" json:"someOtherField"`
}

func (m *Article) TableName() string {
	return "articles"
}


...

a := &Article{
	Title: Lorem ipsum,
	Active: true,
	SomeOtherField: 123
}

if err := app.Dao().Save(a); err != nil {
	return err
}
```
(_note that this may not work if you rely on the `OnModel*` hooks as the `e.Model` will be your custom model_)

**Reply by ganigeorgiev** [reply]
Another a little hacky workaround could be also to temp remove the related record collection schema fields from the model before the db save, something like:

```go
func SaveRecordFields(dao *daos.Dao, record *models.Record, fields ...string) error {
	modelSchema := record.Collection().Schema

	tempRemovedFields := []*schema.SchemaField{}

	// temp remove other fields
	for _, f := range modelSchema.Fields() {
		if list.ExistInSlice(f.Name, fields) {
			continue // keep the field
		}

		modelSchema.RemoveField(f.Id)
		tempRemovedFields = append(tempRemovedFields, f)
	}

	if err := dao.SaveRecord(record); err != nil {
		return err
	}

	// restore removed fields
	for _, f := range tempRemovedFields {
		modelSchema.AddField(f)
	}

	return nil
}

...

record.Set("title", "Lorem ipsum")
record.Set("active", true)

if err := SaveRecordFields(app.Dao(), record, "title", "active"); err != nil {
	return err
}
```

I've added in my todo for future version to consider supporting optional fields in `SaveRecord()` as variadic argument similar to the above.

**Comment by wandomPewlin** [comment]
For people coming after me. It's a lot easier to achieve this now. Just call `rec.IgnoreUnchangedFields(true)` before you save the record, and things should be sorted out for you automatically.
